### PR TITLE
CM_APP_40 | Update profile password API

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

File: src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java [hunk 1]
```diff
@@ -4,23 +4,15 @@
...
-import in.projecteka.consentmanager.user.model.Identifier;
-import in.projecteka.consentmanager.user.model.IdentifierType;
-import in.projecteka.consentmanager.user.model.OtpMediumType;
-import in.projecteka.consentmanager.user.model.OtpVerification;
-import in.projecteka.consentmanager.user.model.Profile;
-import in.projecteka.consentmanager.user.model.SignUpSession;
-import in.projecteka.consentmanager.user.model.Token;
-import in.projecteka.consentmanager.user.model.UpdateUserRequest;
-import in.projecteka.consentmanager.user.model.UserSignUpEnquiry;
+import in.projecteka.consentmanager.user.model.*;
```
Comment (Veena-tw) [inline src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java:15]: do not collapse all imports. Change intellij settings:
Preferences -> Editor -> Code Style -> Java -> Imports
Class count to use import with *: **999**
Use single class import

File: src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java [hunk 2]
```diff
@@ -274,4 +269,105 @@ public void shouldReturnErrorWhenPwdIsInvalidWhileUpdating() {
...
+    }
+
+    @Test
+    public void shouldReturnErrorForInvalidPasswordUpdateRequest() {
+        UpdatePasswordRequest request = UpdatePasswordRequest.builder()
+                .oldPassword("Test@1234")
+                .newPassword("Test")
+                .build();
+        String userName = "user@ncg";
+        Session expectedSession = Session.builder()
```
Comment (Veena-tw) [inline src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java:309]: is this needed?

File: src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java [hunk 3]
```diff
@@ -274,4 +269,105 @@ public void shouldReturnErrorWhenPwdIsInvalidWhileUpdating() {
...
+                .header(AUTHORIZATION, token)
+                .accept(MediaType.APPLICATION_JSON)
+                .bodyValue(request)
+                .exchange()
+                .expectStatus()
+                .is5xxServerError();
+    }
+
+    @Test
+    public void shouldReturnErrorIfOldPasswordValidationFails() {
```
Comment (Veena-tw) [inline src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java:352]: tests the case where the oldpassword given is incorrect.

File: src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java [hunk 4]
```diff
@@ -377,4 +371,72 @@ public void updateUserDetailsFailsWhenTokenForAdminFails() {
...
+                .oldPassword("password@09")
+                .build();
+        Session oldSession = session().build();
+        Session newSession = session().build();
+        Session tokenForAdmin = session().build();
+        String token = String.format("Bearer %s", tokenForAdmin.getAccessToken());
+        KeyCloakUserRepresentation userRepresentation = KeyCloakUserRepresentation.builder()
+                .id("keycloakuserid")
+                .build();
+        when(tokenService.tokenForUser(userName, request.getOldPassword())).thenReturn(Mono.just(oldSession));
```
Comment (Veena-tw) [inline src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java:388]: verify all the mocks used

File: src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java [hunk 5]
```diff
@@ -377,4 +371,72 @@ public void updateUserDetailsFailsWhenTokenForAdminFails() {
...
+        String userName = "user@ncg";
+        UpdatePasswordRequest request = UpdatePasswordRequest.builder()
+                .oldPassword("Test@3142")
+                .newPassword("Test@1234")
+                .build();
+        var oldSession = session().build();
+        var newSession = session().build();
+
+        when(tokenService.tokenForUser(userName, request.getOldPassword())).thenReturn(Mono.just(oldSession));
+        when(tokenService.tokenForAdmin()).thenReturn(Mono.error(ClientError.failedToUpdatePassword()));
```
Comment (Veena-tw) [inline src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java:431]: why will tokenForAdmin return failed to update password error?

File: src/main/java/in/projecteka/consentmanager/user/PatientsController.java [hunk 6]
```diff
@@ -5,19 +5,7 @@
...
-import in.projecteka.consentmanager.user.model.IdentifierType;
-import in.projecteka.consentmanager.user.model.OtpMediumType;
-import in.projecteka.consentmanager.user.model.OtpVerification;
-import in.projecteka.consentmanager.user.model.Profile;
-import in.projecteka.consentmanager.user.model.SignUpRequest;
-import in.projecteka.consentmanager.user.model.Token;
-import in.projecteka.consentmanager.user.model.UpdateUserRequest;
-import in.projecteka.consentmanager.user.model.UserSignUpEnquiry;
-import in.projecteka.consentmanager.user.model.ValidatePinRequest;
+import in.projecteka.consentmanager.user.model.*;
```
Comment (Veena-tw) [inline src/main/java/in/projecteka/consentmanager/user/PatientsController.java:8]: do not collapse imports. modify in all files

File: src/main/java/in/projecteka/consentmanager/user/UserService.java [hunk 7]
```diff
@@ -124,18 +114,27 @@
...
     }
 
-    private Mono<Session> updatedSessionFor(UpdateUserRequest updateUserRequest, String userName) {
+
+    public Mono<Session> updatePasswordFor(String password, String userName) {
+        return getSession(password, userName, failedToUpdatePassword());
+    }
+
+    private Mono<Session> updatedSessionFor(String password, String userName) {
+        return getSession(password, userName, failedToUpdateUser());
```
Comment (Veena-tw) [inline src/main/java/in/projecteka/consentmanager/user/UserService.java:121]: why send the error as parameter, it is already a static import...use it directly where it is needed

File: src/main/java/in/projecteka/consentmanager/user/UserService.java [hunk 8]
```diff
@@ -124,18 +114,27 @@
...
+    }
+
+    private Mono<Session> getSession(String password, String userName, ClientError clientError) {
```
Comment (Veena-tw) [inline src/main/java/in/projecteka/consentmanager/user/UserService.java:130]: there is an update of user and session with the new creds...do you want to rename?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5515255&bid=18072267)

File: src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java (hunk 2 above)
Comment (Saisree9) [reply src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java]: removed

File: src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java (hunk 3 above)
Comment (Saisree9) [reply src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java]: Updated

File: src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java (hunk 4 above)
Comment (Saisree9) [reply src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java:438]: Done

File: src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java (hunk 5 above)
Comment (Saisree9) [reply src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java]: Updated

File: src/main/java/in/projecteka/consentmanager/user/UserService.java (hunk 7 above)
Comment (Saisree9) [reply src/main/java/in/projecteka/consentmanager/user/UserService.java]: Updated

File: src/main/java/in/projecteka/consentmanager/user/UserService.java (hunk 8 above)
Comment (Saisree9) [reply src/main/java/in/projecteka/consentmanager/user/UserService.java]: Updated

**Comment by Saisree9** [comment]
@Veena-tw Updated the PR with review changes. Please approve it.

**Pull request merged**